README.md

```diff
@@ -0,0 +1,148 @@
+
+### 1. `/embed`
+To generate and store embeddings for a GitHub repository.
+
+#### Parameters
+The parameters are passed as a JSON object in the request body:
+
+- `owner` (string, required): The owner of the repository.
+- `name` (string, required): The name of the repository.
+- `branch` (string, required): The name of the branch.
+
+#### Response
+The request is processed by the server and responses are sent as [Server-sent events(SSE)](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events). The event stream will contain the following events with optional data. https://github.com/open-sauced/repo-query/blob/f2f415a4fa9c02d4530624fd7bac2105eea1a77c/src/routes/events.rs#L14-L20
+
+#### Example
+```bash
+curl --location 'localhost:3000/embed' \
+--header 'Content-Type: application/json' \
+--data '{
+    "owner": "open-sauced",
+    "name": "ai",
+    "branch": "beta"
+}'
+```
+
+### 2. `/query`
+To perform a query on the API with a specific question related to a repository.
+
+#### Parameters
+The parameters are passed as a JSON object in the request body:
+
+- `query` (string, required): The question or query you want to ask.
+- `repository` (object, required): Information about the repository for which you want to get the answer.
+  - `owner` (string, required): The owner of the repository.
+  - `name` (string, required): The name of the repository.
+  - `branch` (string, required): The name of the branch.
+
+#### Response
+The request is processed by the server and responses are sent as [Server-sent events(SSE)](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events). The event stream will contain the following events with optional data. https://github.com/open-sauced/repo-query/blob/f2f415a4fa9c02d4530624fd7bac2105eea1a77c/src/routes/events.rs#L22-L29
+
+
+#### Example
+```bash
+curl --location 'localhost:3000/query' \
+--header 'Content-Type: application/json' \
+--data '{
+    "query": "How is the PR description being generated using AI?",
+    "repository": {
+        "owner": "open-sauced",
+        "name": "ai",
+        "branch": "beta"
+    }
+}'
+```
+
+### 3. `/collection`
+To check if a repository has been indexed.
+
+#### Parameters
+- `owner` (string, required): The owner of the repository.
+- `name` (string, required): The name of the repository.
+- `branch` (string, required): The name of the branch.
+
+#### Response
+This endpoint returns an `OK` status code if the repository has been indexed by the service.
+
+#### Example
+```bash
+curl --location 'localhost:3000/embed?owner=open-sauced&name=ai&branch=beta'
+```
+
+## 🧪 Running Locally
+
+To run the project locally, there are a few prerequisites:
+
+- The [Rust toolchain](https://www.rust-lang.org/learn/get-started)
+- The [Onnx Runtime](https://onnxruntime.ai/docs/install/). Will be downloaded and installed automatically when building the project.
```

Comment:
Can you point me to where this automatic install / setup is happening? It wasn't immediately clear to me from my first pass on the code.

Reply:
The `ort` crate does it for us.
https://github.com/pykeio/ort#strategies